File: ems-frontend/src/services/ShiftService.js
import axios from 'axios';

const SHIFT_REST_API_BASE_URL = 'http://localhost:8081/api/shifts';

export const createShift = (shift) => axios.post(SHIFT_REST_API_BASE_URL, shift);

export const getShiftById = (shiftId) => axios.get(SHIFT_REST_API_BASE_URL + '/' + shiftId);

export const getAllShifts = () => axios.get(SHIFT_REST_API_BASE_URL);

export const updateShift = (shiftId, shift) => axios.put(SHIFT_REST_API_BASE_URL + '/' + shiftId, shift);

export const deleteShift = (shiftId) => axios.delete(SHIFT_REST_API_BASE_URL + '/' + shiftId);

export const assignEmployeeToShift = (shiftId, employeeId) => 
    axios.put(`${SHIFT_REST_API_BASE_URL}/${shiftId}/employees/${employeeId}`);

export const removeEmployeeFromShift = (shiftId, employeeId) => 
    axios.delete(`${SHIFT_REST_API_BASE_URL}/${shiftId}/employees/${employeeId}`);

File: ems-backend/src/main/java/net/javaguides/ems/service/ShiftService.java
package net.javaguides.ems.service;

import net.javaguides.ems.dto.ShiftDto;
import java.util.List;

public interface ShiftService {
    ShiftDto createShift(ShiftDto shiftDto);
    
    ShiftDto getShiftById(Long shiftId);
    
    List<ShiftDto> getAllShifts();
    
    ShiftDto updateShift(Long shiftId, ShiftDto shiftDto);
    
    void deleteShift(Long shiftId);
    
    ShiftDto assignEmployeeToShift(Long shiftId, Long employeeId);
    
    ShiftDto removeEmployeeFromShift(Long shiftId, Long employeeId);
}